src/app/forms/components/checkbox-group/checkbox-group.component.ts

```diff
@@ -15,33 +15,40 @@ import { BaseControlComponent } from '../base-control/base-control.component';
   ]
 })
 export class CheckboxGroupComponent extends BaseControlComponent {
-  @Input() options: MultiOptions = [];
+  @Input() options: FormNodeOption<string | number | boolean>[] = [];
+  @Input() separator?: string;
 
-  isChecked(option: string) {
-    const checked = this.value && (this.value as Array<string>).includes(option);
+  isChecked(option: string | number | boolean): boolean {
+    const checked = this.value && this.value.includes(option);
     return checked;
   }
 
-  handleChange(event: boolean, option: MultiOption) {
+  handleChange(event: boolean, option: FormNodeOption<string | number | boolean>): void {
     event ? this.add(option) : this.remove(option);
   }
 
-  private add(option: MultiOption) {
-    !this.value ? (this.value = [option.value]) : this.value.push(option.value);
+  private add(option: FormNodeOption<string | number | boolean>) {
+    if (this.separator) {
+      !this.value ? (this.value = option.value) : (this.value += this.separator + option.value);
+    } else {
+      !this.value ? (this.value = [option.value]) : this.value.push(option.value);
+    }
     this.onChange(this.value);
   }
 
-  private remove(option: MultiOption) {
-    if (this.value && this.value.length > 0) {
-      const i = (this.value as string[]).indexOf(option.value);
+  private remove(option: FormNodeOption<string | number | boolean>) {
+    let newValue = this.separator ? this.value.split(this.separator) : [...this.value];
+    if (newValue && newValue.length > 0) {
+      const i = newValue && newValue.indexOf(option.value);
```

Comment:
We already know that `newValue` is populated from the previous check
```suggestion
      const i = newValue.indexOf(option.value);
```